src/main.rs

```diff
@@ -1,161 +1,34 @@
-use arrow::{
-    array::{ArrayRef, Float64Array, StringArray, UInt64Array},
-    datatypes::{DataType, Field, Schema},
-    record_batch::RecordBatch,
-};
-
-use bitcoin::{Denomination, Txid};
-use bitcoincore_rest::{responses::GetMempoolEntryResult, Error, Network, RestApi, RestClient};
-use chrono::Timelike;
-use std::{
-    collections::{HashMap, HashSet},
-    sync::Arc,
-    time::Instant,
-};
-
-use parquet::{arrow::ArrowWriter, basic::Compression, file::properties::WriterProperties};
-
-type Mempool = HashMap<Txid, GetMempoolEntryResult>;
-
-#[tokio::main]
-async fn main() -> Result<(), Error> {
-    let rest_client = RestClient::network_default(Network::Bitcoin);
-
-    let mut prev_height = 0u64;
-    let mut this_height;
-    let mut prev_mempool: Mempool = HashMap::new();
-    let mut this_mempool: Mempool;
-    let mut prev_timestamp = 0i64;
-
-    loop {
-        // execute once per quarter-minute (:00/:15/:30/:45), preventing double execution
-        let now = chrono::Utc::now();
-        if now.second() % 15 != 0 || prev_timestamp == now.timestamp() {
-            tokio::time::sleep(tokio::time::Duration::from_millis(50)).await;
-            continue;
-        }
-
-        // check height
-        this_height = rest_client.get_chain_info().await?.blocks;
-        let is_new_height = prev_height != this_height;
-        if is_new_height {
-            prev_mempool = HashMap::new();
-            println!("new_height: {:?}", this_height);
-        }
-
-        let start = Instant::now();
-        this_mempool = rest_client.get_mempool().await?;
-        let duration = start.elapsed();
-
-        let batch = create_batchrecord_delta_from_pools(&prev_mempool, &this_mempool);
-
-        println!(
-            "batch_num_rows: {:?}, load_mempool_duration_millis: {:?}",
-            batch.num_rows(),
-            duration.as_millis()
-        );
-
-        let mut filename = std::path::PathBuf::new();
-        let day = now.format("%Y/%m/%d").to_string();
-        let timestamp = now.timestamp();
-        let suffix = if is_new_height { "full" } else { "delta" };
-        filename.push("data");
-        filename.extend(day.split('/'));
-        filename.push(format!("{this_height}_{timestamp}_{suffix}.parquet"));
-        save_batchrecord_deltas_to_parquet(batch, filename);
-
-        prev_height = this_height;
-        prev_mempool = this_mempool;
-        prev_timestamp = now.timestamp();
-    }
+use anyhow::Result;
+use clap::{Parser, Subcommand};
+use tracing::metadata::LevelFilter;
+use tracing_subscriber::EnvFilter;
+use wtf::record::Record;
```

Comment:
is it possible to use `crate::...` instead?

Reply:
I think in `main.rs` you cannot...